app/src/main/java/com/example/mykotlincalculator21/mvp/model/CalculatorModel.kt

```diff
@@ -1,52 +1,135 @@
 package com.example.mykotlincalculator21.mvp.model
 
 import com.example.mykotlincalculator21.mvp.contract.CalculatorContract
-import com.example.mykotlincalculator21.utils.EMPTY_STRING
-import com.example.mykotlincalculator21.utils.ERROR_MESSAGE
-import com.example.mykotlincalculator21.utils.OPERATOR_DIVIDE
-import com.example.mykotlincalculator21.utils.OPERATOR_MINUS
-import com.example.mykotlincalculator21.utils.OPERATOR_MULTIPLY
-import com.example.mykotlincalculator21.utils.OPERATOR_SUM
+import com.example.mykotlincalculator21.utils.ErrorUtils
+import com.example.mykotlincalculator21.utils.NumbersUtils.POSITION_ONE
+import com.example.mykotlincalculator21.utils.NumbersUtils.POSITION_ZERO
+import com.example.mykotlincalculator21.utils.NumbersUtils.ZERO_NUMBER_DOUBLE_TYPE
+import com.example.mykotlincalculator21.utils.OperandUtils
+import com.example.mykotlincalculator21.utils.StringUtils.EMPTY_STRING
+import com.example.mykotlincalculator21.utils.StringUtils.OPERATOR_DIVIDE
+import com.example.mykotlincalculator21.utils.StringUtils.OPERATOR_MINUS
+import com.example.mykotlincalculator21.utils.StringUtils.OPERATOR_MULTIPLY
+import com.example.mykotlincalculator21.utils.StringUtils.OPERATOR_SUM
 
 class CalculatorModel : CalculatorContract.CalculatorModelContract {
-    private var firstOperand: String = EMPTY_STRING
-    private var secondOperand: String = EMPTY_STRING
+    private val firstOperandUtils = OperandUtils()
+    private val secondOperandUtils = OperandUtils()
     private var operator: String = EMPTY_STRING
     private var result: String = EMPTY_STRING
+    private var error = ErrorUtils.NONE
 
     override fun saveNumber(number: String) {
         if (operator.isEmpty()) {
-            firstOperand += number
+            firstOperandUtils.addNumber(number)
         } else {
-            secondOperand += number
+            secondOperandUtils.addNumber(number)
         }
     }
 
-    override fun getPartialResult() = "$firstOperand$operator$secondOperand"
+    fun getValue(): String {
+        var firstValue = (firstOperandUtils.getValue()).toString()
+        var secondValue = (secondOperandUtils.getValue()).toString()
+        if (secondOperandUtils.isEmpty()) {
+            secondValue = EMPTY_STRING
+        }
+        return "$firstValue$operator$secondValue"
+    }
 
-    override fun getFullResult(): String {
-        var firstOperandValue = firstOperand.toDouble()
-        var secondOperandValue = secondOperand.toDouble()
+    override fun getPartialResult(): String {
```

Comment:
this is necessary? you can directly call getValue, and don't do this. getPartialResult didn't have other logic only call other method so you can call directly getValue

Reply:
Thx for the tip, now its done